### PR TITLE
When adding markers: use initial location (instead of later when saved)

### PR DESCRIPTION
**Pull request by MissCaldin**
I have worked on the issue "When adding markers: use initial location (instead of later when saved)".

My code should create the matker when the activity of editing the marker were launched, without the information about the marker's title, type and description being filled. For that, I add a code to create a marker in the function onCreate() of the class MarkerEditActivity. I use the function addMarker of the class TrackRecordingServiceConnection because it has access to the location of the device. 

Link to the issue : https://github.com/OpenTracksApp/OpenTracks/issues/1393

Many thanks for your help,

Sophie

### CODE REVIEW

**Comment by dennisguse** [comment]
What's your plan if the add operation is canceled by the user?

File: src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java [hunk 1]
```diff
@@ -121,6 +121,9 @@ protected void onCreate(Bundle savedInstanceState) {
         });
 
         viewModel = new ViewModelProvider(this).get(MarkerEditViewModel.class);
+
+        viewModel.getTrackRecordingServiceConnection().addMarker(getApplication(), null, null,null,null);
```
Comment (dennisguse) [inline src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java:125]: This always creates a new marker even if the user only wants to edit an existing one.

File: build.gradle [hunk 2]
```diff
@@ -6,7 +6,7 @@ buildscript {
         mavenCentral()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:8.2.0-beta06'
+        classpath 'com.android.tools.build:gradle:8.2.0-rc01'
```
[the remark this answers is not in the document]
Comment (MissCaldin) [reply build.gradle]: It doesn't work when I put beta06, so, in my next commit I put "beta06", but I have dont my tests with rc01

File: src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java (hunk 1 above)
Comment (MissCaldin) [reply src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java]: I add a if condition

**Comment by MissCaldin** [comment]
I was also wondering : Are MarkerEditActivity and MarkerEditViewModel launched at the same time ? If so, maybe loadData, which has get the marker.Id is launched before my new marker is added by onCreate from MarkerEditActivity. Then, the marker which is created at the end is the one with the new location.

File: build.gradle (hunk 2 above)
[the remark this answers is not in the document]
Comment (dennisguse) [reply build.gradle]: In the main branch, we already use the 8.2.0 stable version.
So, you have to `git rebase main` (and before pull the changes from the upstream git repo).
After you did this, you can use the stable version of AndroidStudio (i.e., Hedgehog).

PS/ The line should look like this afterwards:  classpath 'com.android.tools.build:gradle:8.2.0`

**Comment by dennisguse** [comment]
Superseeded by #1825.

**Pull request closed without merging**